modules/imgproc/src/hal_replacement.hpp

```diff
@@ -615,6 +615,43 @@ inline int hal_ni_integral(int depth, int sdepth, int sqdepth, const uchar * src
 #define cv_hal_integral hal_ni_integral
 //! @endcond
 
+/**
+   @brief Calculate box filter
+   @param src_depth,dst_depth Depths of source and destination image
+   @param src_data,src_step Source image
+   @param dst_data,dst_step Destination image
+   @param width,height Source image dimensions
+   @param cn Number of channels
+   @param margins Margins for source image
+   @param ksize Size of kernel
+   @param anchor Anchor point
+   @param normalize If true then result is normalized
+   @param border_type Border type
+*/
+inline int hal_ni_boxFilter(int src_depth, int dst_depth, const uchar* src_data, size_t src_step, uchar* dst_data, size_t dst_step, int width, int height, int cn, CvRect margins, CvSize ksize, CvPoint anchor, bool normalize, int border_type) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
```

Comment:
We tried to avoid all complex data structs in HAL interface, please use only types defined int `hal/interface.h` headers.
Do we need all provided values to run the filter?

Reply:
Yes, I need all values. Some of them needs only for to checking, I can check that anchor == (-1, -1) in OpenCV but what if another HAL could work with anchor?

Reply:
It will be better to split each function into `Init`, `run` and `Free` stages, like it is done for _Filter2d_, _SepFilter2d_ and others.